open-cluster-management/backlog#11100 Upd prep sub

cadawson opened the pull request.

cadawson committed:
open-cluster-management/backlog#11100 Upd prep sub

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
+* Method 1 - Managed cluster was created using {product-title-short}
 +
-You can use the `SubmarinerConfig` API to build the cluster environment. With this method, the `submariner-addon` configures the environment, so you use your configurations and cloud provider credentials in your `SubmarinerConfig` definition. 
+Using this method, you apply the `SubmarinerConfig` API to the {product-title-short} hub cluster. When you run this procedure and do not specify an IPSec IKE port or IPSec NATT port, the default ports are automatically opened for the communication. The default ports are (500) and (4500), respectively. 
```
I think we may need the type of ports, e.g. The default ports are (500/UDP) and (4500/UDP)

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
+Replace `<managed-cluster-namespace>` with the namespace of your managed cluster.
++
+Replace `<managed-cluster-name>-aws-creds` with your AWS credential secret, which you can find in the cluster namespace of your hub cluster. 
++
```
I think we may only need to replace `<managed-cluster-name>`, because the suffix `-aws-creds` is fixed

skeeey replied:
Replace `managed-cluster-name` with your managed cluster name, the name `<managed-cluster-name>-aws-creds` is your AWS credential secret name, which you can find in the cluster namespace of your hub cluster. 

the following are the same

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `<managed-cluster-namespace>` with the namespace of your managed cluster.
++
+Replace `<managed-cluster-name>-aws-creds` with your AWS credential secret, which you can find in the cluster namespace of your hub cluster. 
```
I think we may only need to replace `<managed-cluster-name>`, because the suffix `-aws-creds` is fixed

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `<managed-cluster-name>-aws-creds` with your AWS credential secret, which you can find in the cluster namespace of your hub cluster. 
++
+Replace `IKEPort` with the port that you want to use for your Internet Key Exchange port.
```
Replace `<IKEPort>` ?

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `IKEPort` with the port that you want to use for your Internet Key Exchange port.
++
+Replace `NATTPort` with the port that you want to use for your network address translation - traversal port.
```
Replace `<IKEPort>`?
Replace `<NATTPort>` ?

cadawson replied:
You are correct that I needed to make this consistent. I used these as a basis for the others. I removed the < and > from the other ones outside of the example. We discussed how to approach the style on our squad, and thought this would be a good way. Thanks!

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
 ----
 +
-The format of `_name_` is the same as the credential secret name that you used to provision the cluster with {product-title-short}.
+Replace `<aws-credential-secret-name>`` with the name of your credential secret file.
```
there are two apostrophes, it should be  `<aws-credential-secret-name>`

cadawson replied:
Yes, there were some places that I added one, and VSCode kindly added one, as well. Thank you.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
-The format of `_name_` is the same as the credential secret name that you used to provision the cluster with {product-title-short}.
+Replace `<aws-credential-secret-name>`` with the name of your credential secret file.
++
+Replace `<managed-cluster-namespace>`` with the namespace of your managed cluster.
```
there are two apostrophes, it should be  `<managed-cluster-namespace>`

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
+Using this method, you apply the `SubmarinerConfig` API to the {product-title-short} hub cluster. When you run this procedure and do not specify an IPSec IKE port or IPSec NATT port, the default ports are automatically opened for the communication. The default ports are (500) and (4500), respectively. 
 +
-*Note:* This method is only supported when the cluster is on Amazon Web Services, and not on other environments. 
+** If you want to use the default ports for `IPSecIKEPort` (port 500) and `IPSecNATTPort` (port 4500), create a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced.

cadawson replied:
Ok. I changed this.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
++
+** If you want to customize your Internet Key Exchange (IKE) port or network address translation - traversal (NATT) port, create a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `<aws-secret-access-key>` with your AWS access key.
+
+. Apply the `SubmarinerConfig` on your {product-title-short} hub cluster by creating a file that is similar to the following file:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+** If you want to customize your IKE port or NATT port, complete the following steps:
++
+. Create a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++ 
+Replace `<managed-cluster-namespace>` with the namespace of your managed cluster.
++
+Replace `IKEPort` with the port that you want to use for your IKE port.
```
Replace `<IKEPort>` ?

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
 +
-You can use the script file, `prep_for_subm.sh`, that is provided on the Submariner website to update your {ocp-short} installer-provisioned Amazon Web Services infrastructure for Submariner deployments.
-See https://submariner.io/getting-started/quickstart/openshift/aws/#prepare-aws-clusters-for-submariner[Prepare AWS Clusters for Submariner] for the script and instructions for running it.
+Replace `NATTPort` with the port that you want to use for your NATT port.
```
Replace `< NATTPort >`?

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
+Replace `<aws-secret-access-key>` with your AWS access key.
 
-* Method 2    
+. Apply the `SubmarinerConfig` on your {product-title-short} hub cluster by creating a file that is similar to the following file:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
+
+* Method 1 - Managed cluster was created using {product-title-short}
++
+Using this method, you apply the `SubmarinerConfig` API to the {product-title-short} hub cluster. When you run this procedure and do not specify an IPSec IKE port or IPSec NATT port, the default ports are automatically opened for the communication. The default ports are (500) and (4500), respectively.
```
I think we may need the type of ports, e.g. The default ports are (500/UDP) and (4500/UDP)

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+. Label at least one worker node on your managed cluster with the label: `submariner.io/gateway=true`.
++
+. Create a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `<managed-cluster-namespace>` with the namespace of your managed cluster.
++
+Replace `<managed-cluster-name>-gcp-creds` with your Google Cloud Platform credential secret, which you can find in the cluster namespace of your hub cluster. 
```
we may only need to replace `<managed-cluster-name>`, because the suffix `-gcp-creds` is fixed

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
 +
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
+
+** If you want to customize your Internet Key Exchange (IKE) port or network address translation - traversal (NATT) port, create a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `<managed-cluster-namespace>` with the namespace of your managed cluster.
++
+Replace `IKEPort` with the port that you want to use for your Internet Key Exchange port.
```
`<IKEPort>` ?

dockerymick replied:
Replace `IKEPort` with the IKE port that you want to use.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `IKEPort` with the port that you want to use for your Internet Key Exchange port.
++
+Replace `NATTPort` with the port that you want to use for your network address translation - traversal port.
```
`<NATTPort>` ?

dockerymick replied:
Replace `NATTPort` with the NATT port that you want to use.

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `NATTPort` with the port that you want to use for your network address translation - traversal port.
++
+Replace `<managed-cluster-name>-gcp-creds` with your AWS credential secret, which you can find in the cluster namespace of your hub cluster. 
```
we may only need to replace `<managed-cluster-name>`, because the suffix `-gcp-creds` is fixed

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+** If you want to use the default ports for `IPSecIKEPort` (port 500) and `IPSecNATTPort` (port 4500), complete the following steps:
++
+. Create a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
-Replace `_name_` with your rule name.
-Replace `_network-name_` with your Google Cloud Platform cluster network name.
++
+Replace `<gcp-credential-secret-name>`` with the name of your credential secret file.
```
there are two apostrophes, it should be `<gcp-credential-secret-name>`

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `<gcp-credential-secret-name>`` with the name of your credential secret file.
++
+Replace `<managed-cluster-namespace>`` with the namespace of your managed cluster.
```
there are two apostrophes, it should be `<managed-cluster-namespace>`

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
+Replace `<gcp-os-service-account-json-file-content>` with the contents of your Google Cloud Platform `osServiceAccount.json` file.
 
-. Create the inbound and outbound firewall rules on your Google Cloud Platform to open the 8080/TCP port to export metrics service from the Submariner Gateway nodes:
+. Apply the `SubmarinerConfig` on your {product-title-short} hub cluster by creating a file that is similar to the following file:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+** If you want to customize your IKE port or NATT port, complete the following steps:
++
+. Create a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `<gcp-os-service-account-json-file-content>` with the contents of your Google Cloud Platform `osServiceAccount.json` file.
+
+. Apply the `SubmarinerConfig` on your {product-title-short} hub cluster by creating a file that is similar to the following file:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++ 
+Replace `<managed-cluster-namespace>` with the namespace of your managed cluster.
++
+Replace `IKEPort` with the port that you want to use for your IKE port.
```
`<IKEPort>`?

dockerymick replied:
similar update as previously suggested

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,338 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `IKEPort` with the port that you want to use for your IKE port.
++
+Replace `NATTPort` with the port that you want to use for your NATT port.
```
`< NATTPort >`?

dockerymick replied:
Same here

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
 [#preparing-vm-bare]
 === Preparing to deploy Submariner on VMware vSphere or bare metal
 
+Submariner uses IP security (IPSec) to establish the secure tunnels between the clusters on the gateway nodes. It uses virtual extensible LAN (VXLAN) to encapsulate traffic when it moves from the worker and master nodes to the gateway nodes. When you run this procedure and do not specify an IPSec IKE port or IPSec VXLAN port, the default ports are automatically opened for the communication. The d ...
```
@cadawson the vsphere is very different other platforms (aws, gcp, etc), for vsphere,  these ports are not opened by submariner-addon automatically, they are required to be opened on corporate firewalls by vsphere administrator manually in advance

There are three ports are need to be opened
- For IPSec, the ports are `IPSecIKEPort` (500/UDP) and `IPSecNATTPort` (4500/UDP), these two ports are required to access outside cluster and can be customized, and user can customize these two ports, e.g. user can open ports 501 for IPSecIKEPort and 4501 for IPSecNATTPort instead of default ports, after this, user need to use SubmarinerConfig to tell submariner-addon which ports they are used
- For VXLAN, the port is 4800/UDP, the port should be fixed, use cannot change it, and it only need to access inside the clusters

cadawson replied:
Thank you for the clarification here. That helps. I made some changes that I think should cover it.

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
 To prepare the VMware vSphere and bare metal clusters for deploying Submariner, complete the following steps:
 
-. Configure a publicly routable IP address that is designated to the gateway node on at least one of the clusters. 
+* If you want to use the default ports for `IPSecIKEPort` (port 500) and `VXLANPort` (port 4800), complete the following steps:
```
if user use the default ports, submariner-addon knows the default ports, so we don't need these parts

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
+
+* If you want to customize your IKE port or VXLAN port, complete the following steps:
```
How about: if you customized your IPSec ports, complete the following steps:

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+* If you want to customize your IKE port or VXLAN port, complete the following steps:
 
-. Ensure that ports for IP security are open. The default ports are 4500/UDP and 500/UDP. If the default ports are blocked by a firewall, configure a pair of custom ports that are available, like 4501/UDP and 501/UDP.
+. Create and run a file that contains information that is similar to the following example:
```
the following example is not a file, it is a command that can be run directly after the variables are replaced

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+    namespace: <managed-cluster-namespace>
+spec:
+    IPSecIKEPort: <IKEPort>
+    IPSecVXLANPort: <VXLANPort>
```
It should be `IPSecNATTPort `: `<nattPort>`

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+spec:
+    IPSecIKEPort: <IKEPort>
+    IPSecVXLANPort: <VXLANPort>
+    credentialsSecret:
```
For vshpere, we don't need this field

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
++
+Replace `<managed-cluster-namespace>` with the namespace of your managed cluster.
++
+Replace `IKEPort` with the port that you want to use for your Internet Key Exchange port.
```
`<ikePort>`?

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
++
+Replace `IKEPort` with the port that you want to use for your Internet Key Exchange port.
++
+Replace `VXLANPort` with the port that you want to use for your VXLAN port.
```
It should be 
Replace `<NATTPort>` with the port that you want to use for your network address translation - traversal port.

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
++
+Replace `VXLANPort` with the port that you want to use for your VXLAN port.
++
+Replace `<managed-cluster-name>-vmware-creds` with your VMware credential secret, which you can find in the cluster namespace of your hub cluster. 
```
We don't need this for vsphere

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
++
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
+
+. Label at least one worker node on your managed cluster with the label: `submariner.io/gateway=true`.
```
We don't need this

skeeey commented on services/submariner.adoc:
```diff
@@ -283,8 +524,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+
+. Label at least one worker node on your managed cluster with the label: `submariner.io/gateway=true`.
+
+. Create a file that contains information that is similar to the following example:
```
We don't need this for vsphere

cadawson committed:
open-cluster-management/backlog#11100 rev

cadawson committed:
open-cluster-management/backlog#11100 rev cmts

cadawson commented:
Additional changes made. Please take a look @skeeey . Thanks!

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,355 @@ Ensure that you have the following prerequisites before using Submariner:
 * Firewall configuration across all nodes in each of the managed clusters must allow 4800/UDP in both directions. 
 +
-*Note:* This is configured automatically when your clusters are deployed in an Amazon Web Services environment, but must be configured manually for clusters on other environments and for the firewalls that protect private clouds like bare metal and VMware vSphere. 
+*Note:* This is configured automatically when your clusters are deployed in an AWS, Google Cloud Platform, VMware vSphere, or bare metal environment, but must be configured manually for clusters on other environments and for the firewalls that protect private clouds. 
```
The platform does not include `VMware vSphere` and  `bare metal`, and we may move this to line `#45`, because for AWS and GCP, all required ports can be configured automatically by submariner-addon, for other platforms, user need to configure them manually

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,355 @@ Ensure that you have the following prerequisites before using Submariner:
++
+** If you want to use the default ports for `IPSecIKEPort` and `IPSecNATTPort`, complete the following steps:
++
+. Label at least one worker node on your managed cluster with the label: `submariner.io/gateway=true`.
```
We can remove this, after the story #11580,  the submariner-addon will label the worker node automatically

skeeey commented on services/submariner.adoc:
```diff
@@ -38,97 +38,355 @@ Ensure that you have the following prerequisites before using Submariner:
++
+Replace `managed-cluster-namespace` with the namespace of your managed cluster.
++
+Replace `managed-cluster-name` with your Google Cloud Platform credential secret, which you can find in the cluster namespace of your hub cluster. 
```
Replace managed-cluster-name with your managed cluster name, the name <managed-cluster-name>-gcp-creds is your Google Cloud Platform credential secret name, which you can find in the cluster namespace of your hub cluster.

the following are the same

skeeey commented on services/submariner.adoc:
```diff
@@ -280,11 +538,85 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+
+* If you want to use the default ports for `IPSecIKEPort` and `NATTPort`, complete the following steps:
++
+. Label at least one worker node on your managed cluster with the label: `submariner.io/gateway=true`.
```
We can remove this, after the story #11580, the submariner-addon will label the worker node automatically

cadawson replied:
Same with line 508, I guess: . Label a node that has the public gateway as `submariner.io/gateway=true` in the cluster.? Remove that, too?

skeeey replied:
for line 508, it's a manual operator, we need keep it, for azure, ibm, we also need to add this step, but I think we can keep this on this pr for now, because we will still update this part,  I think we can update them later.

skeeey commented on services/submariner.adoc:
```diff
@@ -280,11 +538,85 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+metadata:
+    name: submariner
+    namespace: <managed-cluster-namespace>
+spec:
```
spec: {}

skeeey commented:
@cadawson, looks good, I left a few comments, and  I noticed the `Prerequisites` part, we also need to update the OCP version of hub

> A {product-title-short} hub cluster that is running on {ocp} version 4.5, or later, with Kubernetes version 1.17, or later.

A {product-title-short} hub cluster that is running on {ocp} version 4.6, or later, with Kubernetes version 1.19, or later.

thank you

cadawson committed:
open-cluster-management/backlog#11100 Rev 2

cadawson commented:
Hello @skeeey. I added the updates: One question about removing line 508. We removed the other similar ones, so I wanted to make sure that this one should also be removed. Other than that, I think we should be in good shape. Thank you!

cadawson commented:
Hi @skeeey . Are we waiting on this one for updates, as well, or was that only for the deployment issue? If so, that is ok. If not, can you take a look at this, when you have a chance? I thought you might have thought this one was the other PR that is open at the same time. The other PR is the one that we talked about waiting until the changes are complete in the code. Let me know if it should be both. Thanks!

skeeey commented:
@cadawson, this pr is not dependent on the  deployment part, so we can merge this pr firstly

We introduce some new configurations in story #11580 and GUI team also update the ACM GUI to support this, so we still need to update this part, and because we have too many updates on this pr :),  I think we can merge this pr firstly, then we do other changes based on this pr, I approved this pr, if you feel fine, please merger this, thank you

cadawson commented:
I agree with your plan, @skeeey . As long as the updates will be smaller changes based on this information (rather than based on the original information), let's move forward with your plan. Thank you!

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
 
-* A {product-title-short} hub cluster that is running on {ocp} version 4.5, or later, with Kubernetes version 1.17, or later.
+* A {product-title-short} hub cluster that is running on {ocp} version 4.6, or later, with Kubernetes version 1.19, or later.
 * A credential for accessing the hub cluster with `cluster administrator` permissions.
```
A credential to access the hub cluster with `cluster-admin` permissions

OR 

A credential to access the hub cluster with cluster administrator permissions.

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
 * Firewall configuration across all nodes in each of the managed clusters must allow 4800/UDP in both directions. 
-+
-*Note:* This is configured automatically when your clusters are deployed in an Amazon Web Services environment, but must be configured manually for clusters on other environments and for the firewalls that protect private clouds like bare metal and VMware vSphere. 
 * Firewall configuration on the Gateway nodes allows ingress 8080/TCP so the other nodes in the cluster can access it. 
```
should this be:

Firewall configuration on the Gateway nodes must allow ingress 8080/TCP so the other nodes in the cluster can access it.

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
-*Note:* This is configured automatically when your clusters are deployed in an Amazon Web Services environment, but must be configured manually for clusters on other environments and for the firewalls that protect private clouds like bare metal and VMware vSphere. 
 * Firewall configuration on the Gateway nodes allows ingress 8080/TCP so the other nodes in the cluster can access it. 
-* Firewall configuration open for UDP/4500, UDP/500, and any other ports that are used for IPSec traffic on the gateway nodes.
+* Firewall configuration open for 4500/UDP, 500/UDP, and any other ports that are used for IPSec traffic on the gateway nodes (configured automatically for AWS and Google Cloud Platform).
```
Is this statement in parenthesis necessary since you mention it in the note??

cadawson replied:
Probably not. I was making sure that they were aware of it, but might be too much.

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
 == Preparing the hosts to deploy Submariner
 
-Before deploying Submariner with {product-title}, you must prepare the clusters on the hosting environment for the connection. The requirements vary by the hosting environment, so follow the instructions for your hosting environment.
+Before deploying Submariner with {product-title}, you must prepare the clusters on the hosting environment for the connection. The requirements vary for different hosting environments, so follow the instructions for your hosting environment.
```
Before you deploy Submariner with .....

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
 === Preparing Amazon Web Services to deploy Submariner
 
-There are two ways that you can configure the {ocp-short} cluster that is hosted on Amazon Web Services to integrate with a Submariner deployment. Select one of these options to prepare for the connection:
+You can use the `SubmarinerConfig` API to configure the Amazon Web Services (AWS) cluster to integrate with a Submariner deployment. There are two ways that you configure the {ocp-short} cluster that is hosted on AWS to integrate it with a Submariner deployment, depending on how the cluster was created. The first procedure applies if you created the cluster by using {product-title-short}. The sec ...
```
Since you wrote out the entire  cluster provider previously in this file, I recommend writing:

You can use the `SubmarinerConfig` API to configure the AWS cluster to integrate with a Submariner deployment. There are two ways for you to configure {ocp-short}...

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
-There are two ways that you can configure the {ocp-short} cluster that is hosted on Amazon Web Services to integrate with a Submariner deployment. Select one of these options to prepare for the connection:
+You can use the `SubmarinerConfig` API to configure the Amazon Web Services (AWS) cluster to integrate with a Submariner deployment. There are two ways that you configure the {ocp-short} cluster that is hosted on AWS to integrate it with a Submariner deployment, depending on how the cluster was created. The first procedure applies if you created the cluster by using {product-title-short}. The sec ...
+
+* Method 1 - Managed cluster was created using {product-title-short}
```
I would add a colon at the end of this

cadawson replied:
Not sure about adding a colon here. Would like it to be more of a title than an intro.

dockerymick replied:
Gotchu, maybe use the H4 heading or bold it?

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
+
+* Method 1 - Managed cluster was created using {product-title-short}
++
+Using this method, you apply the `SubmarinerConfig` API to the {product-title-short} hub cluster. When you run this procedure and do not specify an IPSec IKE port or IPSec NATT port, the default ports are automatically opened for the communication. The default ports are (500/UDP) and (4500/UDP), respectively. 
```
Are the ports displayed with the parenthesis or were you trying to bring emphasis to it?

cadawson replied:
I don't think I need them here. I will leave them in other places where I am setting it apart, but will remove them.

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
++
+Replace `IKEPort` with the port that you want to use for your Internet Key Exchange port.
++
+Replace `NATTPort` with the port that you want to use for your network address translation - traversal port.
```
I wouldn't write out the entire name here, just write:

Replace `IKEPort` with the IKE port that you want to use.

Replace `NATTPort` with the NATT port that you want to use.

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
 
-* Method 1 
+* Method 2 - Managed cluster was not created with {product-title-short}   
```
I suggest adding a colon after {product-title-short}

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
 To prepare the clusters on your Google Cloud Platform for deploying the Submariner component, complete the following steps:
 
-. Create the inbound and outbound firewall rules on your Google Cloud Platform to open the IPsec IKE (by default 500/UDP) and NAT traversal ports (by default 4500/UDP) to enable Submariner communication:
+You can use the `SubmarinerConfig` API to configure the Google Cloud Platform cluster to integrate with a Submariner deployment. There are two ways that you configure the {ocp-short} cluster that is hosted on Google Cloud Platform to integrate with a Submariner deployment, depending on how the cluster was created. The first procedure applies if you created the cluster by using {product-title-shor ...
```
There are two ways for you to configure the OCP...

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
-. Create the inbound and outbound firewall rules on your Google Cloud Platform to open the IPsec IKE (by default 500/UDP) and NAT traversal ports (by default 4500/UDP) to enable Submariner communication:
+You can use the `SubmarinerConfig` API to configure the Google Cloud Platform cluster to integrate with a Submariner deployment. There are two ways that you configure the {ocp-short} cluster that is hosted on Google Cloud Platform to integrate with a Submariner deployment, depending on how the cluster was created. The first procedure applies if you created the cluster by using {product-title-shor ...
+
+* Method 1 - Managed cluster was created using {product-title-short}
```
Add a colon after the conref

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
+
+* Method 1 - Managed cluster was created using {product-title-short}
++
+Using this method, you apply the `SubmarinerConfig` API to the {product-title-short} hub cluster. When you run this procedure and do not specify an IPSec IKE port or IPSec NATT port, the default ports are automatically opened for the communication. The default ports are (500/UDP) and (4500/UDP), respectively.
```
Instead of parenthesis, maybe have the port values monospaced

dockerymick commented on services/submariner.adoc:
```diff
@@ -24,111 +24,367 @@ You can enable Submariner on the {ocp-short} clusters that are hosted in the fol
+*Note:* The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
 
-. Create the inbound and outbound firewall rules on your Google Cloud Platform to open the 4800/UDP port to encapsulate Pod traffic from the worker and master nodes to the Submariner Gateway nodes:
+** If you want to customize your Internet Key Exchange (IKE) port or network address translation - traversal (NATT) port, enter a command that contains information that is similar to the following example:
```
Only acronym is needed here

dockerymick commented on services/submariner.adoc:
```diff
@@ -152,12 +408,12 @@ $ az network nic ip-config inbound-nat-rule add \
-Replace `_nic-name_` with your network interface card (NIC) name.
+Replace `lb-name` with your load balancer name.
+Replace `res-group` with your resource group name.
+Replace `nat-name` with your load balancing inbound NAT rule name.
```
Replace `nat-name` with the NAT rule name of the load balancing inbound.

cadawson replied:
Sounded like this was a little confusing, so I reworded it.

dockerymick commented on services/submariner.adoc:
```diff
@@ -280,11 +536,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+[#preparing-vm]
+=== Preparing to deploy Submariner on VMware vSphere
+
+Submariner uses IP security (IPSec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPSec ports, or you can specify other IPSec ports that you configured. When you run this procedure without specifying an IPSec IKE port or an IPSec NATT port, the default ports are automatically used for the communication. The default ports are (500/UDP) an ...
```
As mentioned in the previous suggestion, are the parenthesis necessary here?

dockerymick commented on services/submariner.adoc:
```diff
@@ -280,11 +536,83 @@ To configure the Google Cloud Platform clusters on Red Hat OpenShift Dedicated,
+
+Submariner uses IP security (IPSec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPSec ports, or you can specify other IPSec ports that you configured. When you run this procedure without specifying an IPSec IKE port or an IPSec NATT port, the default ports are automatically used for the communication. The default ports are (500/UDP) an ...
+
+Submariner uses virtual extensible LAN (VXLAN) to encapsulate traffic when it moves from the worker and master nodes to the gateway nodes on the .The VXLAN port cannot be customized, and is always port 4800/UDP.
```
should "nodes on the " be removed?

cadawson committed:
open-cluster-management/backlog#11100 peer

cadawson commented:
Thanks for the review @dockerymick !

dockerymick commented on services/submariner.adoc:
```diff
@@ -499,7 +499,7 @@ There are two kinds of Red Hat OpenShift Kubernetes Service (ROKS) on IBM Cloud:
 To configure the ROKS clusters on a VPC to use Submariner, complete the steps in the following links:
 
-. Before creating a cluster, specify subnets for pods and services, which avoids overlapping CIDRs with other clusters. Make sure there are no overlapping pods and services CIDRs between clusters if you are using an existing cluster.See https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-subnets#vpc_basics[VPC Subnets] for the procedure. 
+. Before creating a cluster, specify subnets for pods and services, which avoids overlapping CIDRs with other clusters. Make sure there are no overlapping pods and services CIDRs between clusters if you are using an existing cluster. See https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-subnets#vpc_basics[VPC Subnets] for the procedure. 
```
Before you create a cluster

dockerymick reviewed (approved):
Thanks for your explanation Chris! I left two more comments, otherwise the changes look good

cadawson committed:
open-cluster-management/backlog#11100 peer 2
